pkgs/development/libraries/opencv/3.x.nix

```diff
@@ -110,8 +94,12 @@ stdenv.mkDerivation rec {
     (opencvFlag "OPENEXR" enableEXR)
     (opencvFlag "CUDA" enableCuda)
     (opencvFlag "CUBLAS" enableCuda)
-  ] ++ lib.optionals enableContrib [ "-DOPENCV_EXTRA_MODULES_PATH=${contribSrc}/modules" ]
-    ++ lib.optionals enableCuda [ "-DCUDA_FAST_MATH=ON" ];
+  ] ++ lib.optionals enableContrib [
+    "-DOPENCV_EXTRA_MODULES_PATH=${contribSrc}/modules"
+    # the following modules are badly behaved and attempt to download 3rd party libs as part of the build process:
+    "-DBUILD_opencv_dnn=OFF"          # attempts to download protobuf src and googlenet caffe model
```

Comment:
I find cmake "syntax" fairly impossible to comprehend at the best of times and there is some odd stuff in here...

Interestingly, building with `-DBUILD_PROTOBUF=OFF` seems to have the effect of disabling both the protobuf download *and* the googlenet model download. Which is odd. It appears the googlenet model is only used in the tests, which we seem to build but not run. This is probably because the tests require quite a large download of external data. This suggests the option of disabling the test building - it would be nice to run them, but we don't, and building them just feels like a waste of time & space.

I'm going to crawl back into bed with some paracetamol anyone who feels like having a go ad untangling this mess is most welcome.